cmd/log.go

```diff
@@ -0,0 +1,46 @@
+package cmd
+
+import (
+	"os"
+
+	"github.com/redhat-developer/odo/pkg/application"
+	"github.com/redhat-developer/odo/pkg/component"
+	"github.com/redhat-developer/odo/pkg/project"
+	"github.com/spf13/cobra"
+)
+
+var logCmd = &cobra.Command{
+	Use:   "log [component_name]",
+	Short: "Retrieve the log for the given component.",
+	Long:  `Retrieve the log for the given component.`,
+	Example: `  # Get the logs for the nodejs component,
+  odo log nodejs
+	`,
+	Args: cobra.RangeArgs(0, 1),
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// Retrieve stdout / io.Writer
+		stdout := os.Stdout
+
+		// Retrieve the client
+		client := getOcClient()
+
+		// Application
+		currentApplication, err := application.GetCurrent(client)
+		checkError(err, "")
+
+		// Project
+		currentProject := project.GetCurrent(client)
+
+		// Retrieve and set the currentComponent
+		currentComponent := getComponent(client, args[0], currentApplication, currentProject)
```

Comment:
It is failing here if no argument is passed to `odo log` command.

If there is no argument, you have to pass an empty string as `inputComponent` (second) parameter and `getComponent` will return the name of the current component. If `inputComponent`  parameter is non-empty string it will check that `inputComponent`  exists and return it back.